### PR TITLE
pandoc-crossref: Source from GitHub instead of Hackage

### LINKED ISSUE #360: v0.3.13.0b missing from Hackage
v0.3.13.0a and v0.3.13.0b are missing from https://hackage.haskell.org/package/pandoc-crossref, which is where Homebrew gets the package it uses to build the app. I'd like to bump the Homebrew formula (in Homebrew/homebrew-core#109242) to this new version in order to get pandoc 2.19 support and silence some warnings.

Even a v0.3.13.1 would suffice for my need.

### LINKED ISSUE #367: pandoc-crossref 0.3.13.0 incompatible with pandoc 2.19.2 on homebrew
pandoc-crossref 0.3.13.0 as binary-installed over homebrew appears incompatible with pandoc 2.19.2:

> WARNING: pandoc-crossref was compiled with pandoc 2.18 but is being run through 2.19.2. This is not supported. Strange things may (and likely will) happen silently.

This happens even though https://formulae.brew.sh/formula/pandoc-crossref#default says that it depends on panda 2.19.2

### PR DESCRIPTION
**Pull request by colindean**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

As discussed in lierdakil/pandoc-crossref#360, Hackage metadata updates don't propagate into the source archives its hosts, so dependency bump versions don't create new versions that Homebrew's tooling can see. This is a running issue with pandoc-crossref, that updates necessary for it to keep up with pandoc releases causing some confusion (e.g. lierdakil/pandoc-crossref#367). #118436 pulled in b3c5f71a which should nudge a human to bump the revision field in the formulae, but that's inconvenient and requires a human.

@lierdakil [suggested][1] using the GitHub source archive instead of Hackage and [again][2]. I [suggested it][3], too, so let's do that.

The hash of the source archive is different because the Hackage archive doesn't have the metadata files that the GitHub archive has. Otherwise, the source is the same.

[1]: https://github.com/lierdakil/pandoc-crossref/issues/360#issuecomment-1289764832
[2]: https://github.com/lierdakil/pandoc-crossref/issues/367#issuecomment-1345630540
[3]: https://github.com/Homebrew/homebrew-core/issues/117921#issuecomment-1356827775

Closes lierdakil/pandoc-crossref#360
Closes lierdakil/pandoc-crossref#367

### CODE REVIEW

**Comment by SMillerDev** [comment]
I think a revision bump is needed here

File: Formula/pandoc-crossref.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class PandocCrossref < Formula
   desc "Pandoc filter for numbering and cross-referencing"
   homepage "https://github.com/lierdakil/pandoc-crossref"
-  url "https://hackage.haskell.org/package/pandoc-crossref-0.3.15.0/pandoc-crossref-0.3.15.0.tar.gz"
-  sha256 "b9d8b6ed246fcac6955e14d38b229b62b0682b98ec2caf7fdc095438f16dfd4f"
+  url "https://github.com/lierdakil/pandoc-crossref/archive/v0.3.15.0.tar.gz"
+  sha256 "d0780e08bae9b589eaad01be1fc818ab551f7debe26a9d1be57aed58032a6b5e"
   license "GPL-2.0-or-later"
```
Comment (colindean) [inline Formula/pandoc-crossref.rb:6]:
```suggestion
  license "GPL-2.0-or-later"
  revision 1
```

Comment (colindean) [reply Formula/pandoc-crossref.rb:6]: Odd that I can't merge my own suggestion onto this branch…

Comment (colindean) [reply Formula/pandoc-crossref.rb:6]: Oh, looks like the perms got adjusted and I can't make changes on this branch now. @SMillerDev Can you merge that suggestion?

**Comment by lierdakil** [comment]
The fact this doesn't build is likely my fault, I didn't yet update to pandoc 3.1 (likely just need to relax dependency constraints).

**Comment by lierdakil** [comment]
v0.3.15.1 is out with support for pandoc 3.1 (the version bump is due to a bugfix)

**Comment by p-linnane** [comment]
https://github.com/Homebrew/homebrew-core/pull/124221

**Pull request closed without merging**